arangod/Cluster/AgencyCache.h

```diff
@@ -211,7 +214,8 @@ class AgencyCache final : public ServerThread<ArangodServer> {
 
   /// @brief Waiting room for indexes during office hours
   mutable std::mutex _waitLock;
-  std::multimap<consensus::index_t, futures::Promise<arangodb::Result>>
+  std::multimap<consensus::index_t,
+                std::pair<futures::Promise<arangodb::Result>, Executor>>
```

Comment:
Maybe define a struct instead of a pair? You can still use structured bindings with structs.

Reply:
Good idea, done.